lang/syn/src/parser/accounts/constraints.rs

```diff
@@ -464,43 +464,36 @@ impl<'ty> ConstraintGroupBuilder<'ty> {
 
         // Token.
         if let Some(token_mint) = &self.token_mint {
-            if self.token_authority.is_none() {
+            if self.init.is_some() && self.token_authority.is_none() {
                 return Err(ParseError::new(
                     token_mint.span(),
-                    "token authority must be provided if token mint is",
-                ));
-            }
-
-            if self.init.is_none() {
-                return Err(ParseError::new(
-                    token_mint.span(),
-                    "init is required for a pda token",
+                    "token authority must be provided if token mint is when initialize",
                 ));
             }
         }
         if let Some(token_authority) = &self.token_authority {
-            if self.token_mint.is_none() {
+            if self.init.is_some() && self.token_mint.is_none() {
                 return Err(ParseError::new(
                     token_authority.span(),
-                    "token mint must be provided if token authority is",
+                    "token mint must be provided if token authority is when initialize",
                 ));
             }
         }
 
         // Mint.
         if let Some(mint_decimals) = &self.mint_decimals {
-            if self.mint_authority.is_none() {
+            if self.init.is_some() && self.mint_authority.is_none() {
                 return Err(ParseError::new(
                     mint_decimals.span(),
-                    "mint authority must be provided if mint decimals is",
+                    "mint authority must be provided if mint decimals is when initialize",
                 ));
             }
         }
         if let Some(mint_authority) = &self.mint_authority {
-            if self.mint_decimals.is_none() {
+            if self.init.is_some() && self.mint_decimals.is_none() {
                 return Err(ParseError::new(
                     mint_authority.span(),
-                    "mint decimals must be provided if mint authority is",
+                    "mint decimals must be provided if mint authority is when initialize",
                 ));
```

Comment:
since these are now only thrown when `init.is_some()`, what do you think about moving all of these checks into the `// Init` section?

Reply:
Hmm, that's a good idea.